plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts

```diff
@@ -119,6 +121,181 @@ export function createRouter({
+    }
+
+    let argoAppResp = {};
+    try {
+      argoAppResp = await argoSvc.createArgoApplication(
+        matchedArgoInstance.url,
+        token,
+        projectName,
+        appName,
+        namespace,
+        sourceRepo,
+        sourcePath,
+        labelValue,
+      );
+      return response.send({
+        argoProjectName: projectName,
+        argoAppName: appName,
+        kubernetesNamespace: namespace,
+      });
+    } catch (e: any) {
+      logger.error(argoAppResp);
+      return response.status(500).send({
+        status: 500,
+        message: e.message || 'Failed to create argo app',
+      });
+    }
+  });
+
+  router.post('/sync', async (request, response) => {
+    const appSelector = request.body.appSelector;
+    const argoSvc = new ArgoService(argoUserName, argoPassword, config);
+    try {
+      const argoSyncResp = await argoSvc.resyncAppOnAllArgos(appSelector);
+
+      return response.send(argoSyncResp);
+    } catch (e: any) {
+      return response.status(e.status || 500).send({
+        status: e.status || 500,
+        message: e.message || `Failed to sync your app, ${appSelector}.`,
+      });
+    }
+  });
+
+  router.delete(
+    '/argoInstance/:argoInstanceName/applications/:argoAppName',
+    async (request, response) => {
+      const argoInstanceName = request.params.argoInstanceName;
+      const argoAppName = request.params.argoAppName;
+      logger.info(`Getting info on ${argoInstanceName} and ${argoAppName}`);
+
+      const argoSvc = new ArgoService(argoUserName, argoPassword, config);
+
+      const matchedArgoInstance = argoInstanceArray.find(
+        argoInstance => argoInstance.name === argoInstanceName,
+      );
+      if (matchedArgoInstance === undefined) {
+        return response.status(500).send({
+          status: 'failed',
+          message: 'cannot find an argo instance to match this cluster',
+        });
+      }
+
+      let token: string;
+      if (!matchedArgoInstance.token) {
+        token = await argoSvc.getArgoToken(matchedArgoInstance.url);
+      } else {
+        token = matchedArgoInstance.token;
+      }
+
+      let argoDeleteAppResp: boolean;
+      try {
+        argoDeleteAppResp = await argoSvc.deleteApp(
+          matchedArgoInstance.url,
+          argoAppName,
+          token,
+        );
+      } catch (e :any) {
+        if (
+          typeof(e.message) === "string"
+        ) {
+          throw new Error(
+            e.message
+          );
+        }
+        return response
+          .status(500)
+          .send({ status: 'error with deleteing argo app' });
+      }
+
+      let argoDeleteProjectResp: boolean;
+      try {
+        let argoApp = await argoSvc.getArgoAppData(
+          matchedArgoInstance.url,
+          matchedArgoInstance.name,
+          { name: argoAppName },
+          token,
+        );
+        let isAppDeployed = 'metadata' in argoApp;
+        for (let attempts = 0; attempts < 25 && isAppDeployed; attempts++) {
```

Comment:
Would this ever take longer than 75 seconds on larger apps? Maybe we could make this an optional config value just in case

Reply:
Feels like it shouldn't however I added this as an optional configuration item `waitCycles`